rpcs3/Emu/Cell/SPUThread.cpp

```diff
@@ -3944,19 +3944,23 @@ s64 spu_thread::get_ch_value(u32 ch)
 
 		state += cpu_flag::wait;
 
-		using resrv_ptr = std::add_pointer_t<decltype(rdata)>;
+		using resrv_ptr = std::add_pointer_t<const decltype(rdata)>;
 	
-		resrv_ptr resrv_mem{};
+		resrv_ptr resrv_mem = vm::get_super_ptr<decltype(rdata)>(raddr);
 		std::shared_ptr<utils::shm> rdata_shm;
 
-		if (raddr && mask1 & SPU_EVENT_LR)
+		// Does not need to safe-access reservation if LR is the only event masked
+		// Because it's either an access violation or a livelock if an invalid memory is passed
+		if (raddr && mask1 > SPU_EVENT_LR)
```

Comment:
Why '>'?

Reply:
Well, I suppose it works as of now.

Reply:
On real PS3 reservations are bound to physical memory. Deallocations do no invalidate physical memory thus no LR event is raised. If TM or other events are set it will wait for them on realhw in such case.

Reply:
I was referring to usage of '>' operator as a shortcut of testing for LR+any other flag